extensions/positron-r/src/provider.ts

```diff
@@ -46,70 +52,15 @@ export enum RRuntimeSource {
  * @param context The extension context.
  */
 export async function* rRuntimeDiscoverer(): AsyncGenerator<positron.LanguageRuntimeMetadata> {
-	const binaries = new Map<string, RBinary>();
-
-	function updateBinaries(binary: RBinary | RBinary[] | undefined): void {
-		if (!binary) {
-			return;
-		}
-		const input = Array.isArray(binary) ? binary : [binary];
-		for (const binary of input) {
-			if (binaries.has(binary.path)) {
-				const reasons = new Set(binaries.get(binary.path)!.reasons.concat(binary.reasons));
-				binaries.get(binary.path)!.reasons = Array.from(reasons);
-			} else {
-				binaries.set(binary.path, binary);
-			}
-		}
-	}
-
-	// Find the current R binary(ies) for various definitions of "current".
-	// The first item here will eventually be marked as The Current R Binary.
-	const currentBinaries = await currentRBinaryCandidates();
-	updateBinaries(currentBinaries);
-
-	// Now we scour the system for all R binaries we can find.
-	const systemHqBinaries = discoverHQBinaries(rHeadquarters());
-	updateBinaries(systemHqBinaries);
-
-	const registryBinaries = await discoverRegistryBinaries();
-	updateBinaries(registryBinaries);
-
-	const moreBinaries = discoverAdHocBinaries([
-		'/usr/bin/R',
-		'/usr/local/bin/R',
-		'/opt/local/bin/R',
-		'/opt/homebrew/bin/R'
-	]);
-	updateBinaries(moreBinaries);
-
-	// Optional, user-specified root directories or binaries
-	const userBinaries = discoverUserSpecifiedBinaries();
-	updateBinaries(userBinaries);
-
-	// Search locations for R binaries that are conventional on servers. These locations are also
-	// searched by RStudio/Posit Workbench on POSIX platforms, such as Linux and macOS.
-	// See https://github.com/rstudio/rstudio/blob/bb8cbf17bb415467f87d6e415f9e3777fa46e583/src/cpp/core/r_util/RVersionsPosix.cpp#L121-L147
-	if (os.platform() !== 'win32') {
-		const serverBinaries = discoverServerBinaries();
-		updateBinaries(serverBinaries);
-	}
-
-	// (Try to) promote each RBinary to a proper RInstallation
-	let rInstallations: Array<RInstallation> = [];
-	if (currentBinaries.length > 0) {
-		const currentBinary = currentBinaries[0];
-		rInstallations.push(new RInstallation(currentBinary.path, true, currentBinary.reasons));
-		binaries.delete(currentBinary.path);
-	}
+	// Discover R binaries on the system
+	const { binaries, currentBinary } = await getBinaries();
 
-	binaries.forEach(rbin => {
-		rInstallations.push(new RInstallation(rbin.path, false, rbin.reasons));
-	});
-
-	const rejectedRInstallations: Array<RInstallation> = [];
-	rInstallations = rInstallations
+	// Promote R binaries to R installations
+	const rejectedRInstallations: RInstallation[] = [];
+	const rInstallations: RInstallation[] = binaries
+		.map(rbin => new RInstallation(rbin.path, rbin.path === currentBinary, rbin.reasons))
 		.filter(r => {
+			// Filter out any rejected R installations
```

Comment:
Another example of a comment that doesn't seem necessary. I won't point all of these out as it's somewhat subjective, but it would be great if you took a pass at removing some (I think) co-pilot added. In some cases, the code seems pretty self-explanatory already and in others, the new comment feels redundant with more details, in-place comments just below (eg. sorting by version number).